spanner/mutation.go

```diff
@@ -427,24 +429,50 @@ func (m Mutation) proto() (*sppb.Mutation, error) {
 
 // mutationsProto turns a spanner.Mutation array into a sppb.Mutation array,
 // it is convenient for sending batch mutations to Cloud Spanner.
-func mutationsProto(ms []*Mutation) ([]*sppb.Mutation, error) {
+func mutationsProto(ms []*Mutation) ([]*sppb.Mutation, *sppb.Mutation, error) {
+	var selectedMutation *Mutation
+	var nonInsertMutations []*Mutation
+	maxValues := -1
```

Comment:
Why do we need to store seperate maxValues variable?
This will always be equal to selectedMutation.values() if selectedMutation is not null ?